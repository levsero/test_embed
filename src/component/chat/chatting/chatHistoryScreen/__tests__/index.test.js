import { render } from '@testing-library/react'
import React from 'react'
import { Component as ChatHistoryScreen } from '../'
import snapshotDiff from 'snapshot-diff'

jest.mock('component/chat/chatting/HistoryLog')

const renderChatHistoryScreen = (customProps = {}) => {
  const defaultProps = {
    historyLength: 2,
    hasMoreHistory: false,
    historyRequestStatus: 'done',
    isMobile: false,
    allAgents: {},
    showAvatar: false,
    fetchConversationHistory: noop,
    hideZendeskLogo: false,
    chatId: 'id-1012',
    firstMessageTimestamp: 1555030483758,
    fullscreen: false,
    title: 'this is the title'
  }

  const props = {
    ...defaultProps,
    ...customProps
  }

  return render(<ChatHistoryScreen {...props} />)
}

describe('ChatHistoryScreen', () => {
  let result, defaultProps

  describe('when not mobile', () => {
    beforeEach(() => {
      result = renderChatHistoryScreen()
    })

    it('renders ChatHistory', () => {
      expect(result.container).toMatchSnapshot()
    })
  })

  describe('when mobile', () => {
    beforeEach(() => {
      defaultProps = renderChatHistoryScreen()
      result = renderChatHistoryScreen({ isMobile: true })
    })

    it('renders mobile classes', () => {
      expect(snapshotDiff(defaultProps, result, { contextLines: 2 })).toMatchSnapshot()
    })
  })

  describe('when hideZendeskLogo is true', () => {
    beforeEach(() => {
      defaultProps = renderChatHistoryScreen()
      result = renderChatHistoryScreen({ hideZendeskLogo: true })
    })

    it('has no logo and correct margin', () => {
      expect(snapshotDiff(defaultProps, result, { contextLines: 2 })).toMatchSnapshot()
    })
  })
})
