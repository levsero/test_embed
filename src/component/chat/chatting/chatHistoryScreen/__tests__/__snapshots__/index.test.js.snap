// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatHistoryScreen when hideZendeskLogo is true has no logo and correct margin 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ -37,5 +37,5 @@
        </div>
        <div
-         class="content footerMarginWithLogo headerMargin scrollContainerMessagesContentDesktop"
+         class="content footerMargin headerMargin scrollContainerMessagesContentDesktop"
          data-testid="scroll-container-content"
        >
@@ -62,31 +62,4 @@
        />
      </div>
-     <a
-       class="
-       logo
-       formSuccess
-       
-       zendeskLogo 
-     "
-       href="https://www.zendesk.com/chat/live-chat-widget/?utm_source=webwidgetchat&utm_medium=poweredbyzendeskchat&utm_campaign=poweredbyzendesk&utm_term=id-1012&utm_content=testingHost&iref=id-1012&lang="
-       tabindex="-1"
-       target="_blank"
-     >
-       <span
-         class="container icon Icon--zendesk"
-         data-testid="Icon--zendesk"
-         type="Icon--zendesk"
-       >
-         <svg
-           aria-hidden="true"
-           realfilename="widget-icon_zendesk.svg"
-         />
-       </span>
-       <span
-         class="hidden"
-       >
-         zendesk
-       </span>
-     </a>
    </div>
  </div>
`;

exports[`ChatHistoryScreen when mobile renders mobile classes 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ -2,7 +2,6 @@
    <div>
      <div
-       class="container flexContainer desktop containerDesktop"
+       class="container flexContainer mobile"
        data-testid="scroll-container"
-       style="height: 550px;"
      >
        <div
@@ -37,10 +36,10 @@
        </div>
        <div
-         class="content footerMarginWithLogo headerMargin scrollContainerMessagesContentDesktop"
+         class="content footerMarginWithLogo headerMargin scrollContainerMessagesContent scrollContainerMobile contentMobile"
          data-testid="scroll-container-content"
        >
          <div>
            <div
-             class="chatLogContainer"
+             class="chatLogContainer chatLogContainerMobile"
            >
              <div
@@ -67,5 +66,5 @@
        formSuccess
        
-       zendeskLogo 
+       zendeskLogo zendeskLogoChatMobile
      "
        href="https://www.zendesk.com/chat/live-chat-widget/?utm_source=webwidgetchat&utm_medium=poweredbyzendeskchat&utm_campaign=poweredbyzendesk&utm_term=id-1012&utm_content=testingHost&iref=id-1012&lang="
`;

exports[`ChatHistoryScreen when not mobile renders ChatHistory 1`] = `undefined`;
